R/utils.R

```diff
@@ -84,3 +55,199 @@ with_unclassed_to <- function(x, .class = 1:length(class(x)), expr,
   eval(bquote(class(.(x_expr)) <- .(orig_class)), envir = envir)
   out
 }
+
+
+
+#' Capture side effects issued by an evaluated expression
+#'
+#' All messaging condition side effects are captured in the order that they
+#' are issued.
+#'
+#' @param expr an expression to evaluate, capturing output events as they
+#'   are issued
+#' @param env the environment in which \code{expr} should be evaluated,
+#'   defaulting to the calling environment.
+#' @param quoted whether \code{expr} is a quoted object and should be evaluated
+#'   as is, or whether the expression should be captured from the function call.
+#'   Defaults to \code{FALSE}, capturing the passed expression.
+#' @inheritParams base::sink
+#'
+#' @importFrom utils head tail
+#'
```

Comment:
Could you add an example for this function to help future developer use it in other functions?